cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/s3_dataset.spec.js

```diff
@@ -141,6 +147,40 @@ const s3DatasetTestSuite = () => {
           cy.getElementByTestId('docTable').should('be.visible');
           cy.getElementByTestId('docTable').find('tr').should('have.length', 11);
         });
+
+        it('aborts and cancels previous query when new query is started', function () {
+          cy.getElementByTestId(`datasetSelectorButton`).click();
+          cy.getElementByTestId(`datasetSelectorAdvancedButton`).click();
+
+          cy.get(`[title="S3 Connections"]`).click();
+          cy.get(`[title="BasicS3Connection"]`).click();
+          cy.get(`[title="mys3"]`).click();
+          cy.get(`[title="default"]`).click();
+          cy.get(`[title="http_logs"]`).click();
+          cy.getElementByTestId('datasetSelectorNext').click();
+          cy.get(`[class="euiModalHeader__title"]`).should('contain', 'Step 2: Configure data');
+
+          cy.getElementByTestId('advancedSelectorLanguageSelect').select('OpenSearch SQL');
+          cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+          // Need to wait a bit for initial query to start
+          cy.wait(3000);
+
+          cy.intercept('DELETE', `**/${JOBS_API.DELETE}*`).as('cancelRequest');
+          cy.getElementByTestId(`querySubmitButton`).click();
+
+          cy.wait('@cancelRequest').then((interception) => {
+            console.log('interception.request.url:', interception.request.url);
+            // Verify the request had the correct query parameters
+            expect(interception.request.url).to.include('queryId=');
+            expect(interception.request.url).to.include('id=');
+          });
+
+          cy.waitForSearch();
+          cy.get(`[data-test-subj="queryResultCompleteMsg"]`).should('be.visible');
+          cy.getElementByTestId('docTable').should('be.visible');
+          cy.getElementByTestId('docTable').find('tr').should('have.length', 11);
```

Comment:
could we potentially luck out on this being 11? for no matter what table.

would it be better if we modify the query  to be something we know like `SELECT * FROM foo LIMIT 10` then have it be like `LIMIT 20` and then we make sure it is still 10?